src/parser.rs

```diff
@@ -136,12 +150,29 @@ impl UnaryOperator {
     }
 }
 
+impl Parsable for Literal {
+    fn parse(input: &str) -> IResult<&str, Self> {
+        map(i32, |num: i32| -> Literal { Literal::IntegerLiteral(num) })(input)
```

Comment:
FYI: Rust's bread and butter for functional programming are closures, one of which is used here. See more at https://doc.rust-lang.org/book/ch13-01-closures.html